util/ranger/detacher.go

```diff
@@ -364,6 +521,27 @@ func DetachCondAndBuildRangeForIndex(sctx sessionctx.Context, conditions []expre
 	for _, col := range cols {
 		newTpSlice = append(newTpSlice, newFieldType(col.RetType))
 	}
+	// In order to avoid exponential exploding, only expand once here.
+	if restConditions, dnfItems := expandDNF(conditions, cols); restConditions != nil && dnfItems != nil {
```

Comment:
In this branch, it seems you want to expand all conditions like `a AND (b OR c)` to `(a AND b) or (a AND c)`.
But for `a AND (b OR c) AND (d OR e)`, will it be expanded incorrectly to `(a AND b) or (a AND c) ... or (a AND e)`?